core/templates/dev/head/domain/collection/CollectionRightsObjectFactory.js

```diff
@@ -0,0 +1,113 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating and mutating instances of frontend
+ * collection rights domain objects.
+ */
+
+oppia.factory('CollectionRightsObjectFactory', [
+    '$rootScope', '$log',
+    function($rootScope, $log) {
+    var CollectionRights = function(collectionRightsObject) {
+      this._collectionId = collectionRightsObject.collection_id;
+      this._canEdit = collectionRightsObject.can_edit;
+      this._canUnpublish = collectionRightsObject.can_unpublish;
+      $rootScope.isPrivate = collectionRightsObject.is_private;
```

Comment:
It can't be on `$rootScope` -- it logically/conceptually doesn't belong there, and it breaks encapsulation. It does need to be on `this`, and we need to find a way to watch it.

Could you identify, more specifically, why "pointers aren't enough"? Are the two references to the collection domain object actually referencing the same object (and not a copy)? If not, that's a problem, since copies won't sync.